fun/middleware/enforce_legal_acceptance.py

```diff
@@ -44,31 +42,27 @@ def terms_accepted(func):
     AGREEMENT_FORM else defaults to DEFAULT_AGREEMENT_FORM
     """
     def wrapped(request, *args, **kwargs):
-
         if not legal_acceptance(request.user):
-            return redirect(
-                get(settings, "AGREEMENT_FORM", DEFAULT_AGREEMENT_FORM),
-                *args, **kwargs
-            )
+            return redirect(TERMS_AND_CONDITIONS_AGREEMENT, *args, **kwargs)
         else:
             return func(request, *args, **kwargs)
     return wrapped
 
+
 class LegalAcceptance(object):
     def __init__(self, *a, **kw):
-        self.white_list = get(
-            settings,
-            "AGREEMENT_WHITELIST",
-            DEFAULT_AGREEMENT_WHITELIST
-        )
+        self.white_list = AGREEMENT_WHITELIST
 
     def is_whitelisted(self, url):
         return any(wl_url.match(url) for wl_url in self.white_list)
 
     def process_view(self, request, view_func, view_args, view_kwargs):
-        if not self.is_whitelisted(request.path_info) \
-            and hasattr(request, "user") \
-            and request.user.is_authenticated and \
-            not legal_acceptance(request.user):
+        if (
+                not self.is_whitelisted(request.path_info) and \
+                hasattr(request, "user") and \
+                request.user.is_authenticated() and \
+                # don't enforce legal acceptance when impersonating a user
+                not request.user.is_masked and \
+                not legal_acceptance(request.user)):
```

Comment:
I'm not a big fan of this indentation style. What about the following with a comment on top of the block detailing each condition?

```python
# We redirect current request user to the terms acceptance view if:
#   1. the current url is whitelisted
#   2. ...
if not self.is_whitelisted(request.path_info) and \
        hasattr(request, "user") and \
        request.user.is_authenticated() and \
        not request.user.is_masked and \
        not legal_acceptance(request.user):
```